frontend/libretro.c

```diff
@@ -1907,6 +1909,93 @@ void retro_run(void)
          }
       }
 
+		if (in_type[i] == PSE_PAD_TYPE_GUNCON)
+		{
+			//ToDo move across to:
+			//RETRO_DEVICE_ID_LIGHTGUN_SCREEN_X
+			//RETRO_DEVICE_ID_LIGHTGUN_SCREEN_Y   
+			//RETRO_DEVICE_ID_LIGHTGUN_TRIGGER
+			//RETRO_DEVICE_ID_LIGHTGUN_RELOAD
+			//RETRO_DEVICE_ID_LIGHTGUN_AUX_A 
+			//RETRO_DEVICE_ID_LIGHTGUN_AUX_B
+			//Though not sure these are hooked up properly on the Pi
+			
+			//ToDo
+			//Put the controller index back to i instead of hardcoding to 1 when the libretro overlay crash bug is fixed
+			//This is required for 2 player
+			
+			//GUNCON has 3 controls, Trigger,A,B which equal Circle,Start,Cross
+			
+			// Trigger
+			//The 1 is hardcoded instead of i to prevent the overlay mouse button libretro crash bug
+			if (input_state_cb(1, RETRO_DEVICE_MOUSE, 0, RETRO_DEVICE_ID_MOUSE_LEFT)){
+				in_keystate[i] |= (1 << DKEY_CIRCLE);
+			}
+			
+			// A
+			if (input_state_cb(1, RETRO_DEVICE_MOUSE, 0, RETRO_DEVICE_ID_MOUSE_RIGHT)){
+				in_keystate[i] |= (1 << DKEY_START);
+			}
+			
+			// B
+			if (input_state_cb(1, RETRO_DEVICE_MOUSE, 0, RETRO_DEVICE_ID_MOUSE_MIDDLE)){
+				in_keystate[i] |= (1 << DKEY_CROSS);
+			}
+			
+			//The 1 is hardcoded instead of i to prevent the overlay mouse button libretro crash bug
+			int gunx = input_state_cb(1, RETRO_DEVICE_POINTER, 0, RETRO_DEVICE_ID_POINTER_X);
+			int guny = input_state_cb(1, RETRO_DEVICE_POINTER, 0, RETRO_DEVICE_ID_POINTER_Y);
+			
+			//This adjustment process gives the user the ability to manually align the mouse up better 
+			//with where the shots are in the emulator.
+			
+			//Percentage distance of screen to adjust 
+			int GunconAdjustX = 0;
+			int GunconAdjustY = 0;
+			
+			//Used when out by a percentage
+			float GunconAdjustRatioX = 1;
+			float GunconAdjustRatioY = 1;
+				
+			struct retro_variable var;
+   			var.value = NULL;
+   			var.key = "pcsx_rearmed_gunconadjustx";
+   			if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) || var.value)
```

Comment:
its strange that this core is calling RETRO_ENVIRONMENT_GET_VARIABLE within retro_run() and not when RETRO_ENVIRONMENT_GET_VARIABLE_UPDATE return true like its usually done so.

Reply:
> 
> 
> its strange that this core is calling RETRO_ENVIRONMENT_GET_VARIABLE within retro_run() and not when RETRO_ENVIRONMENT_GET_VARIABLE_UPDATE return true like its usually done so.
> 
> this is spamming the logs when guncon is selected.

Hi @negativeExponent, can you clarify if that is something wrong in this core in general or something I did wrong in the code in my pull request.  I still have some further improvements to do for this implementation so can change it to work in the way you suggested.  I'm a beginner so guidance is welcome.

Best regards

SindenLightgun